src/main/java/org/clx/library/controller/AuthorController.java

```diff
@@ -78,69 +18,74 @@ public class AuthorController {
 
     private static final Logger logger = LoggerFactory.getLogger(AuthorController.class);
     private final AuthorService authorService;
+    private static final String MESSAGE = "failed";
 
-    // Create Author Endpoint
     @PostMapping("/createAuthor")
-    public ResponseEntity<String> createAuthor(@RequestBody AuthorRequest authorRequest) {
-        try {
-            // Pass the AuthorDto to the service layer
+    public ResponseEntity<ApiResponse> createAuthor(@RequestBody AuthorRequest authorRequest) {
+            // Pass the AuthorRequest to the service layer
             authorService.createAuthor(authorRequest);
             logger.info("Author created successfully.");
-            return new ResponseEntity<>("Author created successfully", HttpStatus.CREATED);
-        } catch (Exception e) {
-            logger.error("Error creating author: {}", e.getMessage());
-            return new ResponseEntity<>("Failed to create author", HttpStatus.INTERNAL_SERVER_ERROR);
-        }
+
+            // Create a success response
+            ApiResponse response = new ApiResponse(HttpStatus.CREATED,"Author created successfully",null);
+            return new ResponseEntity<>(response, HttpStatus.CREATED);
+
     }
 
-    // Update Author Endpoint by ID
+
     @PutMapping("/updateAuthor/{authorId}")
-    public ResponseEntity<String> updateAuthor(@RequestBody AuthorDto authorDto, @PathVariable Integer authorId) {
+    public ResponseEntity<ApiResponse> updateAuthor(@RequestBody AuthorRequest authorRequest, @PathVariable Integer authorId) {
         logger.info("Received request to update author with ID: {}", authorId);
         try {
-            // Pass the AuthorDto to the service layer along with the ID
-            AuthorDto updatedAuthor = authorService.updateAuthor(authorDto, authorId);
+            AuthorRequest updatedAuthor = authorService.updateAuthor(authorRequest, authorId);
             logger.info("Author with ID: {} updated successfully", authorId);
-            return new ResponseEntity<>("Author updated!!", HttpStatus.ACCEPTED);
-        } catch (AuthorNotFoundException e) {
+
+            ApiResponse response = new ApiResponse(HttpStatus.ACCEPTED,"Author updated successfully", updatedAuthor);
+            return new ResponseEntity<>(response, HttpStatus.ACCEPTED);
+        } catch (ResourceNotFoundException e) {
             logger.error("Author with ID: {} not found", authorId);
-            return new ResponseEntity<>(e.getMessage(), HttpStatus.NOT_FOUND);
-        } catch (Exception e) {
-            logger.error("Error updating author with ID: {}: {}", authorId, e.getMessage());
-            return new ResponseEntity<>("Failed to update author", HttpStatus.INTERNAL_SERVER_ERROR);
+
+            // Create a not-found response
+            ApiResponse response = new ApiResponse(HttpStatus.NOT_FOUND,"Author not found", e.getMessage()
+            );
```

Comment:
exception part expect to throw with general exception

Reply:
now i am understand this thing so we will try to avoid this type to handle exception.